src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java

```diff
@@ -321,11 +287,10 @@ else if (fieldObject instanceof List && fieldJsonElement.isJsonArray()) {
 	 * Add each non-transient additional data property to the given JSON node
 	 *
 	 * @param item the object containing additional data
-	 * @param itemJsonElement	   the JSON node to add the additional data properties to
+	 * @param itemJsonObject	   the JSON node to add the additional data properties to
 	 */
-	private void addAdditionalDataFromJsonElementToJson (final Object item, final JsonElement itemJsonElement) {
-		if (item instanceof IJsonBackedObject && itemJsonElement.isJsonObject()) {
-			final JsonObject itemJsonObject = itemJsonElement.getAsJsonObject();
+	private void addAdditionalDataFromJsonElementToJson (final Object item, final JsonObject itemJsonObject) {
```

Comment:
Consider renaming this from addAdditionalDataFromJsonElementToJson to addAdditionalDataFromJsonObjectToJson

Reply:
done.